### PR TITLE
Show error message correctly in manage storage modal

### LINKED ISSUE #1565: [Bug]: Cannot Update cluster storage size when not bound to a workbench
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Deploy type

Downstream version (eg. `RHODS 1.29`)

### Version

Downstream version - RHODS 1.30.RC3

### Current Behavior

Cannot Update cluster storage size when not bound to a workbench

### Expected Behavior

A Persistent Volume Claim -> Requested capacity with the value filled in Storage Size

### Steps To Reproduce

Create Cluster storage
Edit Cluster storage and type a new value for `Persistent storage size`
The value in `Volume Claim -> Requested capacity` is not changed.

This change is only working if the cluster storage is attached to a workbench

### Workaround (if any)

_No response_

### What browsers are you seeing the problem on?

Chrome

### Anything else

_No response_

### PR DESCRIPTION
**Pull request by DaoDaoNoCode**
Closes #1565 

## Description

As I commented in the original issue, there is nothing we can do with the PVC storage update, but we did wrongly digest the error message which would cause confusion to the users. The way is just to correctly show the error message in the modal.

Screenshot 2023-08-15 at 1 48 46 PM

## How Has This Been Tested?

1. Go to a data science project
2. Create a cluster storage
3. Don't connect it to any notebook, and try to increase the capacity
4. You should be able to see the error message

## Test Impact

N/A, it's a fix to show the network failure call, which is commonly used across all the modals

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: frontend/src/pages/projects/screens/detail/storage/ManageStorageModal.tsx [hunk 1]
```diff
@@ -73,60 +78,49 @@ const ManageStorageModal: React.FC<AddStorageModalProps> = ({ existingData, isOp
...
-      }
-      handleNotebookNameConnection(pvcName);
-      if (parseInt(getPvcTotalSize(existingData)) !== createData.size) {
-        await updatePvcSize(pvcName, namespace, `${createData.size}Gi`);
+        );
       }
+      Promise.all(pvcPromises)
+        .then(() => {
+          if (notebookName) {
+            pvcPromises.push(attachNotebookPVC(notebookName, namespace, pvcName, mountPath.value));
```
Comment (christianvogt) [inline frontend/src/pages/projects/screens/detail/storage/ManageStorageModal.tsx:104]: Shouldn't you return the promise from `attachNotebookPVC` ?
```suggestion
            return attachNotebookPVC(notebookName, namespace, pvcName, mountPath.value);
```

File: frontend/src/pages/projects/screens/detail/storage/ManageStorageModal.tsx [hunk 2]
```diff
@@ -73,60 +78,49 @@ const ManageStorageModal: React.FC<AddStorageModalProps> = ({ existingData, isOp
...
+          }
+        })
+        .then(() => onBeforeClose(true))
+        .catch(handleError);
     } else {
       createPvc(pvc)
-        .then((createdPvc) => handleNotebookNameConnection(createdPvc.metadata.name))
+        .then((createdPvc) => {
+          if (notebookName) {
+            attachNotebookPVC(notebookName, namespace, createdPvc.metadata.name, mountPath.value);
```
Comment (christianvogt) [inline frontend/src/pages/projects/screens/detail/storage/ManageStorageModal.tsx:113]: Same question:
```suggestion
            return attachNotebookPVC(notebookName, namespace, createdPvc.metadata.name, mountPath.value);
```

**Review by christianvogt — CHANGES_REQUESTED** [review]
Error shows up as expected. 

Without performing dry runs, there's now a slightly odd behavior when the user encounters the error when changing the PVC size. If they also changed the name or description along with the PVC size, the error will occur but the name and description changes will have gone through already. If the user now closes the modal, these new values are not reflected in the UI. They continue to see the old values. This wasn't the case before.

Ideally a dry run could be performed for the k8s changes to weed out known errors in the data before performing a full submission. However such changes are a bit beyond the scope of the issue at hand.

**Comment by DaoDaoNoCode** [comment]
Close this PR because the issue will be fixed in #1714 along with the comments above.

**Pull request closed without merging**